bugbug/tools/code_review.py

```diff
@@ -369,8 +369,50 @@ def get_all_inline_comments(
                 yield diff_id, comments
 
 
+class SwarmReviewData(ReviewData):
+    def __init__(self):
+        self.auth = {
+            "user": get_secret("SWARM_USER"),
+            "password": get_secret("SWARM_PASS"),
+            "port": get_secret("SWARM_PORT"),
+            "instance": get_secret("SWARM_INSTANCE"),
+        }
+
+    # return ReviewRequest object with patch_id = revision_id
+    def get_review_request_by_id(self, revision_id: int) -> ReviewRequest:
+        return ReviewRequest(revision_id)
+
+    # return rawdiff from the initial version of the revision with id patch_id
+    def get_patch_by_id(self, patch_id: int) -> Patch:
+        revisions = swarm.get(self.auth, rev_ids=[int(patch_id)], version_l=[0, 1])
+        assert len(revisions) == 1
+        return Patch(revisions[0]["fields"]["diff"])
+
+    # return rawdiff from the initial version of a specific version of the revision_id
+    def get_patch_by_version_fromto(
+        self, revision_id: int, version_before: int = 0, version_num: int = 1
+    ) -> Patch:
+        revisions = swarm.get(
+            self.auth,
+            rev_ids=[int(revision_id)],
+            version_l=[version_before, version_num],
+        )
+        assert len(revisions) == 1
+
+        return Patch(
+            revisions[0]["fields"]["diff"]
+        )
```

Comment:
Why do we need this method?

Reply:
for the addon on swarm we need to support the different versions of the diff, I use this function to get versions later than the first instead of get_patch_by_id. I left the description of this function as it's not part of the Abstract Class and might not apply to github.

Reply:
However, the review tool will only use the methods that exist in the abstract class. I see in I see that in `get_patch_by_id()`, you use `0` to `1` as default values. What is the impact of using just that? Where will the `get_patch_by_version_fromto()` method be used?

Reply:
we had a modified run function in the api call, I added it in the code_review_tool_runner.py now